workspaces/src/operations.rs

```diff
@@ -236,13 +236,28 @@ impl Transaction {
     }
 
     async fn transact_raw(self) -> Result<FinalExecutionOutcomeView> {
-        send_batch_tx_and_retry(
+        let view = send_batch_tx_and_retry(
             self.worker.client(),
             &self.signer,
             &self.receiver_id,
             self.actions?,
         )
-        .await
+        .await?;
+
+        if !self.worker.tx_callbacks.is_empty() {
+            let total_gas_burnt = view.transaction_outcome.outcome.gas_burnt
+                + view
+                    .receipts_outcome
+                    .iter()
+                    .map(|t| t.outcome.gas_burnt)
+                    .sum::<u64>();
+
+            self.worker.tx_callbacks.iter().for_each(|meter| {
+                meter(Gas::from_gas(total_gas_burnt)).unwrap();
```

Comment:
we shouldn't be calling into `unwrap` into all these meter functions. You already have a impl to convert PoisonError into an Error, so why not just return it with `?`

Reply:
Done